### PR TITLE
[drools-4643]Configurable consumers

### PR DESCRIPTION
**Pull request by desmax74**
Configurable poll time for consumers events,control,kiesessioninfo and separately for snapshot

@danielezonca  @jiripetrlik

### CODE REVIEW

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java [hunk 1]
```diff
@@ -198,6 +242,34 @@ public int getMaxSnapshotRequestAttempts() {
...
+    public PollUnit getPollSnapshotUnit() { return pollUnitSnapshot; }
+
+    public Duration getPollDuration(){
+        Duration duration;
+        if(pollUnit == PollUnit.MILLISECOND){
+            duration = Duration.ofMillis(pollTimeout);
+        }else if(pollUnit == PollUnit.SECOND){
+            duration = Duration.ofSeconds(pollTimeout);
+        }else{
+            duration = Duration.ofSeconds(1);
```
Comment (danielezonca) [inline hacep/src/main/java/org/kie/hacep/EnvConfig.java]: Same comment here.
In addition we should try to refactor in a single method this logic, it is the third one with the "same" switch

File: hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java [hunk 2]
```diff
@@ -42,10 +42,10 @@ public void initConsumer( ConsumerHandler consumerHandler ) {
     }
 
     @Override
-    public void poll( int durationMillis ) {
-        String topic = config.getEventsTopicName();
+    public void poll() {
+        String topic = envConfig.getEventsTopicName();
         while (true) {
-            RemoteCommand command = ( RemoteCommand ) queue.poll( topic, durationMillis );
+            RemoteCommand command = ( RemoteCommand ) queue.poll(topic, 1000);
```
Comment (danielezonca) [inline hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java]: Why now it is hardcoded?

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java [hunk 3]
```diff
@@ -128,6 +141,30 @@ public EnvConfig withMaxSnapshotRequestAttempts(String maxSnapshotRequestAttempt
...
     }
 
+    public EnvConfig withPollTimeUnit(String pollTimeUnit){
+        switch (pollTimeUnit){
+            case millisec:
+                this.pollUnit = PollUnit.MILLISECOND;
+            case sec:
+                this.pollUnit = PollUnit.SECOND;
+            default:
+                this.pollUnit = PollUnit.SECOND;
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/EnvConfig.java]: Ok

File: hacep/src/main/java/org/kie/hacep/EnvConfig.java [hunk 4]
```diff
@@ -128,6 +141,30 @@ public EnvConfig withMaxSnapshotRequestAttempts(String maxSnapshotRequestAttempt
...
+    }
+
+    public EnvConfig withPollSnapshotTimeUnit(String pollSnapshotTimeUnit){
+        switch (pollSnapshotTimeUnit){
+            case millisec:
+                this.pollUnitSnapshot = PollUnit.MILLISECOND;
+            case sec:
+                this.pollUnitSnapshot = PollUnit.SECOND;
+            default:
+                this.pollUnitSnapshot = PollUnit.SECOND;
```
[the remark this answers is not in the document]
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/EnvConfig.java]: ok

File: hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java (hunk 2 above)
Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java]: Fixed

Comment (desmax74) [reply hacep/src/main/java/org/kie/hacep/core/infra/consumer/LocalConsumer.java]: Fixed with value from envConfig

**Comment by desmax74** [comment]
Jenkins test this

**Pull request merged**